src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -18,175 +16,167 @@ namespace System.Net.Quic.Implementations.MsQuic
 {
     internal sealed class MsQuicConnection : QuicConnectionProvider
     {
-        private readonly MsQuicSession? _session;
-
-        // Pointer to the underlying connection
-        // TODO replace all IntPtr with SafeHandles
-        private IntPtr _ptr;
+        // Delegate that wraps the static function that will be called when receiving an event.
+        private static readonly ConnectionCallbackDelegate s_connectionDelegate = new ConnectionCallbackDelegate(NativeCallbackHandler);
 
-        // Handle to this object for native callbacks.
-        private GCHandle _handle;
+        // TODO: remove this.
+        // This is only used for client-initiated connections, and isn't needed even then once Connect() has been called.
+        private readonly SafeMsQuicConfigurationHandle? _configuration;
```

Comment:
This is another thing that'd be removable if we switched to a static `Connect` as in a former comment.

Reply:
I consider the TODO as work that needs to be done/resolved. But I don't want to hold the PR on those issues. The work will be done in follow up PRs.